### PR TITLE
Use devtools

### PR DESCRIPTION
**Pull request by MartinSchobben**
I have made some changes here to the Dockerfile to enable installing Devtools. The shell script now helps me to better evaluate the package when build against irods. I can create documentation now, but installing the package (`devtools::build()`) produces an error. It says `-stdlib=libc++` should not be included in the Makevars file.

### CODE REVIEW

File: build-rirods.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# build image
+sudo docker image build --tag rirods:build .
+
+# local source and library paths
+SRCPATH=$(R -e "getwd()"| grep /)
+LIBPATH=$(R -e ".libPaths()[1]" | grep /)
```
Comment (trel) [inline build-rirods.sh]: this requires R is installed on the machine running docker... I was hoping to have little/no dependencies on the host machine.

File: R/RcppExports.R [hunk 2]
```diff
@@ -0,0 +1,276 @@
+# Generated by using Rcpp::compileAttributes() -> do not edit by hand
```
Comment (trel) [inline R/RcppExports.R:1]: if this is a generated file, can we ... not have it in the source?

File: NAMESPACE [hunk 3]
```diff
@@ -1,3 +1,2 @@
-#useDynLib(rirods)
```
Comment (trel) [inline NAMESPACE:1]: if this is a generated file, can we ... not have it in the source?

File: build-rirods.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# build image
+sudo docker image build --tag rirods:build .
```
Comment (trel) [inline build-rirods.sh:4]: shouldn't need sudo in here...

File: build-rirods.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (trel) [inline build-rirods.sh:1]: might rework this to be a Makefile itself...

targets could include...
- dockerimage
- devtoolsdocument
- devtoolsbuild

**Review by trel — COMMENTED** [review]
will run this later today - wrote some initial comments via only inspection, ignore as appropriate :)

File: build-rirods.sh (hunk 1 above)
Comment (MartinSchobben) [reply build-rirods.sh]: My main idea was to help the development of the R package by adding this file (can be deleted later-on). So, for that reason, I use volumes to execute R code (devtools) in the container but have my files updated. This file is also included in the buildignore.

File: R/RcppExports.R (hunk 2 above)
Comment (MartinSchobben) [reply R/RcppExports.R:1]: I would argue that it needs to be included. Then use wrapper function around these automatic generated functions to add the custom R code (which is now included in RccpExportsManual.R) and divide over several R scripts with some logical division of functionality. This is then just for clarity of the code.

File: build-rirods.sh (hunk 5 above)
Comment (MartinSchobben) [reply build-rirods.sh:1]: Custom makefiles are advised against (https://r-pkgs.org/src.html?q=Make#make). I deleted the original Makefile in this branch.

Comment (trel) [reply build-rirods.sh:1]: Yes, I was talking about a Makefile to provide the same functionality as this shell script - just calling docker itself.

No worries, now I read the interaction could be messy.

File: build-rirods.sh (hunk 1 above)
Comment (trel) [reply build-rirods.sh]: Got it - I see.

**Comment by trel** [comment]
We have abandoned this effort - and have decided to deprecate and archive this repository.

Please find a new pure-R implementation at:

https://github.com/irods/irods_client_library_rirods

**Pull request closed without merging**